docs/build/add-attachments.md

```diff
@@ -0,0 +1,215 @@
+---
+sidebar_label: Attachments
+sidebar_position: 3
+description: Support image, video, and document message attachments in your app
+---
+
+# Support message attachments in your app built with XMTP
+
+Use the `RemoteAttachmentCodec` from the `xmtp-content-type-remote-attachment` package to support message attachments in your app—including images, videos, gifs, and documents.
+
+This document provides a step-by-step guide to providing message attachments in your app, as well as key considerations to think about when building this functionality.
+
+For a reference implementation of image attachments, see the [xmtp.chat](http://xmtp.chat) example app. 
+
+## Storage considerations
+
+XMTP messages have a size limit of 1 MB and most attachment types will exceed this size. As a result, you must store attachments in a remote location outside of the XMTP network.
+
+You can use storage options such as Web3 Storage, ThirdWeb Storage, s3 buckets, and others. Another option is to allow your users to provide their own storage token.
+
+The [xmtp.chat](http://xmtp.chat) reference implementation uses Web3 Storage, with XMTP Labs hosting with our own token. We wanted to use a decentralized solution, which made Web3 Storage a good choice. Hosting with our own token reduces user friction, but we cap uploads at 5 MB. In the future, [xmtp.chat](http://xmtp.chat) might give users the option to provide their own token if they want to upload larger file sizes or keep their images beyond a certain timeframe. 
+
+## Security considerations
+
+:::caution
+
+Automatically loading attachments from untrusted users can have negative security implications. Be sure to consider this before enabling attachments to autoload in your app.
+
+:::
+
+A more secure pattern you might want to use is to require the user to click a “Click to load” CTA before loading an attachment. After the attachment has been loaded, it can be pulled from a cache. This pattern requires one more click for the user, but provides greater security.
```

Comment:
Could be good to mention the other privacy tradeoffs with automatically loading attachments, namely that the owner of the server where the attachment is stored will see user information like IP address, user agent, etc.